### PR TITLE
Marine Code sprint 2019-03

### LINKED ISSUE #124: Update marine section of README.md after Marine CS.
The marine section of the README.md should be updated to reflect the addition of various converters during the Marine IODA/UFO code sprint.

### PR DESCRIPTION
**Pull request by travissluka**
marine converters from the march/april 2019 code sprint.
Following changes were made since the `feature/marine-cs-03-2019` branch at the end of the sprint:

- minor cleanup of directory structure, file names, autopep-ing to pass coding norms
- `gds2_sst2ioda.py` can process multiple files in parallel, memory usage improved, code sped up
- documentation of converters on main `README.md` page
- git comit history was rewritten to remove the `src/ncep/xx120` binary file and instead place it on git-lfs

otherwise, all other changes have already been approved on PRs for the `feature/marine-cs-03-2019` branch

(note: future code sprints need to just work on the develop branch.... this was a mess)
closes #124 #104 #103 #91 #90 #86 #87 #88

### CODE REVIEW

File: src/marine/godae/CMakeLists.txt [hunk 1]
```diff
@@ -9,8 +9,3 @@ COPY_FILES( "${programs}" ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_BINARY_DIR}/bin )
 install (PROGRAMS
   ${programs}
   DESTINATION bin)
-
```
Comment (srherbener) [inline src/marine/godae/CMakeLists.txt]: ${programs} list is not getting built properly. The file names are missing the "godae_" prefix.

File: src/marine/hgodas/hgodas_adt2ioda.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
```
Comment (srherbener) [inline src/marine/hgodas/hgodas_adt2ioda.py]: In the file src/marine/hgodas/CMakeLists.txt, the ${programs} list is not getting built properly. The files have underscores in their prefixes, whereas the CMakeLists.txt has hyphens.

File: README.md [hunk 3]
```diff
@@ -37,8 +37,26 @@ See the script tools/build_gsi_nc_files.sh for an example of how to run the flow
...
+* GODAE insitu temperature and salinity ocean profiles from the Fleet Numerical Meteorology and Oceanography Center(FNMOC). Observations available from [here](https://www.usgodae.org/ftp/outgoing/fnmoc/data/ocn/)
+  * `godae_profile2ioda.py`
+  * `godae_ship2ioda.py`
+  * `godae_trak2ioda.py`
+* Hybrid-GODAS - preprocessed, suberobbed, and QCd observations of altimetry, insitu T/S, and SST. Used in the NCEP HGODAS project _(these are likely to be removed at some point)_
+  * `hgodas_adt2ioda.py`
+  * `hgodas_insitu2ioda.py`
+  * `hgodas_sst2ioda.py`  
+* `rads_adt2ioda.py` - absolute dynamic topography observations from NOAA/NESDIS. Observations available from [here] (ftp://ftp.star.nesdis.noaa.gov/pub/sod/lsa/rads/adt)
+* `smap_sss2ioda.py` - SMAP satellite sea surface salinity observations. Observations available from [here](ftp://podaac-ftp.jpl.nasa.gov/allData/smap/L2/RSS/V3/SCI)
```
Comment (aerorahul) [inline README.md]: ftp is not hyperlinked in markdown, I think.

File: README.md [hunk 4]
```diff
@@ -37,8 +37,26 @@ See the script tools/build_gsi_nc_files.sh for an example of how to run the flow
...
+* `gmao_obs2ioda.py` - NASA/GMAO ocean observations
+* GODAE insitu temperature and salinity ocean profiles from the Fleet Numerical Meteorology and Oceanography Center(FNMOC). Observations available from [here](https://www.usgodae.org/ftp/outgoing/fnmoc/data/ocn/)
+  * `godae_profile2ioda.py`
+  * `godae_ship2ioda.py`
+  * `godae_trak2ioda.py`
+* Hybrid-GODAS - preprocessed, suberobbed, and QCd observations of altimetry, insitu T/S, and SST. Used in the NCEP HGODAS project _(these are likely to be removed at some point)_
+  * `hgodas_adt2ioda.py`
+  * `hgodas_insitu2ioda.py`
+  * `hgodas_sst2ioda.py`  
+* `rads_adt2ioda.py` - absolute dynamic topography observations from NOAA/NESDIS. Observations available from [here] (ftp://ftp.star.nesdis.noaa.gov/pub/sod/lsa/rads/adt)
```
Comment (aerorahul) [inline README.md]: ftp is not rendered properly in markdown, I think. Besides, the space between [] and()  is throwing it off.

File: README.md (hunk 3 above)
Comment (travissluka) [reply README.md]: markdown doesn't have a problem with it, but github does for come reason

**Comment by travissluka** [comment]
@aerorahul what was the coding norms complaining about? Everything passes for me, and autopep8 does not produce any changes

**Comment by srherbener** [comment]
@travissluka, I'm seeing errors now when I try rads_adt2ioda.py:
~~~~~~~
> rads_adt2ioda.py -i data/rads_adt_3a_2018105.nc -o test.rads.nc4 -d 2018010500
Traceback (most recent call last):
  File "/usr/local/bin/rads_adt2ioda.py", line 99, in <module>
    writer = iconv.NcWriter(args.o, [], locationKeyList)
AttributeError: 'Namespace' object has no attribute 'o'
~~~~~~~

Since @aerorahul was able to run okay, I'm guessing I've got something out of date or broken in my Python configuration. Looks like an issue with argsparse, but this was working fine yesterday. Any thoughts about what is wrong?

**Comment by travissluka** [comment]
@srherbener my mistake rads_adt2ioda.py and the hgodas_ converters should be fixed now

**Comment by srherbener** [comment]
@travissluka, I'm good with merging now. Are there any other reviewers that you want feedback from first, or do you want to go ahead and merge now?

**Comment by travissluka** [comment]
@srherbener you can go ahead and merge

**Comment by srherbener** [comment]
@travissluka, Done! Can you deleted obsolete branches from the code sprint? If you don't have permission, let me know and I'll delete them. Thanks!

**Comment by srherbener** [comment]
@travissluka, thank you for deleting the obsolete branches!

**Pull request merged**